### DISCUSSION TITLE
Downloading files from Pocketbase Server in Dart/Flutter

### DISCUSSION THREAD

**Discussion by PeetoomHeida** [pocketbase/pocketbase#2240] [Q&A]
I am wondering if it is possible to directly download files in a record in Pocketbase to local storage. I haven't seen anything in the documentation and don't know where to start in implementing possible solutions. The file is a zip file if that makes any difference.

Thanks for your help!

**Comment by ganigeorgiev** [comment]
There is no built-in helper for downloading file (aka. creating an inline `File` instance), but you can use the std http client and just send a regular GET request to the file URL as any other web resource - https://pocketbase.io/docs/files-handling/#file-url.

**Reply by shellking4** [reply]
Ok but please why even doing as you say, the file is zip on mobile client ?

**Reply by ganigeorgiev** [reply]
@shellking4 I'm not sure that I understand what do you mean. The SDKs doesn't have an endpoint for fetching files. We only provide helpers to generate a url to the file. To access/download the file you need to send a GET request to that url.

**Reply by shellking4** [reply]
I'm doing something like below 

```js
export const downloadURI = (uri: string, name: string) => {
  var link = document.createElement("a");
  link.download = name;
  link.href = uri;
  document.body.appendChild(link);
  link.click();
  document.body.removeChild(link);
  return;
}

const download = async () => {
        let record: any
        try {
            record = await pocketbase.collection("constants").getOne("wl2l7sr4dx2mceh");
            downloadURI(`${apiUrlBase}/files/${record?.collectionId}/${record?.id}/${record?.media}`, 'app.apk');
            return;
        } catch (error) {
            console.log(error);
        } finally {
            return record;
        }
    }
  ```

I guess I might be doing it wrong. But when the file is downloaded it is app.apk.zip and I have to remove the .zip

**Reply by ganigeorgiev** [reply]
It seems that you are using JavaScript (the original question was for Dart/Flutter).

The `download` attribute will work only for same-origin URLs (see the note in https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#download). I'm not sure why you have a `.zip` extension. It is possible that it was added by the browser or if the file was uploaded without an extension, the server generated one for it based on its mimetype.

**Reply by shellking4** [reply]
Thank you @ganigeorgiev  I'm using nginx. Could that be caused by some of its configs ? Anyway... don't bother I'm going to send the app on PlayStore soon. Thanks for your great work on pocketbase
